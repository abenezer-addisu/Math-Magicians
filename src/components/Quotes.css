/* Quotes.css */

.quote {
  margin: 20px auto;
  padding: 20px;
  background-color: #f5f5f5;
  border-radius: 5px;
  width: 60%;
  justify-content: center;
}

.quote p {
  margin: 0;
  font-size: 18px;
}
