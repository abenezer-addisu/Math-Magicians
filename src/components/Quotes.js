import React, { useState, useEffect } from 'react';
import './Quotes.css';

const Quotes = () => {
  const [quote, setQuote] = useState({});
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState('');

  useEffect(() => {
    const fetchQuote = async () => {
      try {
        const response = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=happiness',
          {
            headers: {
              'X-Api-Key': 'yLpVtNRgYO0rmsTwWWlZoA==qY5MvAay3LrWTA4Q',
            },
          },
        );
        const data = await response.json();
        const { author, quote } = data[0];
        setQuote({ author, quote });
        setIsLoading(false);
      } catch (error) {
        setError('Failed to fetch the quote.');
        setIsLoading(false);
      }
    };

    fetchQuote();
  }, []);

  return (
    <div className="quote">
      {isLoading && <p>Loading...</p>}
      {error && <p>{error}</p>}
      {!isLoading && !error && (
        <>
          <p className="quote-content">{quote.quote}</p>
          <h2 className="quote-author">{quote.author}</h2>
        </>
      )}
    </div>
  );
};

export default Quotes;
