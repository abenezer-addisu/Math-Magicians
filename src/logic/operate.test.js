import renderer from 'react-test-renderer';
import Quotes from '../components/Quotes';

it('Qoute page is working', () => {
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
