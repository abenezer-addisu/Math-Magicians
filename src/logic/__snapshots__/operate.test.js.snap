// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Qoute page is working 1`] = `
<div
  className="quote"
>
  <p>
    Loading...
  </p>
</div>
`;
